app/(auth)/onboarding.tsx

```diff
@@ -0,0 +1,247 @@
+import Screen from '@src/components/screen';
+import Assets from '@src/theme/assets';
+import screen from '@src/utils/helpers';
+import { Link } from 'expo-router';
+import React, { useEffect, useRef, useState } from 'react';
+import { Animated, FlatList, View } from 'react-native';
+import { Button, XStack, Text, Stack, YStack } from 'tamagui';
+
+interface InformationCarouselContent {
+    title: string;
+    description: string;
+}
+
+const imageCarouselData = [
+    { key: 'spacer-1' },
+    {
+        img: Assets.Girl1(),
+        key: 'item-1',
+    },
+    {
+        img: Assets.Girl2(),
+        key: 'item-2',
+    },
+    {
+        img: Assets.Girl3(),
+        key: 'item-3',
+    },
+    {
+        img: Assets.Girl1(),
+        key: 'item-4',
+    },
+    {
+        img: Assets.Girl2(),
+        key: 'item-5',
+    },
+    {
+        img: Assets.Girl3(),
+        key: 'item-6',
+    },
+    { key: 'spacer-2' },
+];
+
+const duplicateDataWithUniqueKeys = () => {
+    // First, duplicate the array
+    const duplicatedArray = [...imageCarouselData, ...imageCarouselData.slice(1)];
+
+    // Then, modify the keys of the second set
+    return duplicatedArray.map((item, index) => {
+        // If it's part of the second set, modify the key
+        if (index >= imageCarouselData.length) {
+            return { ...item, key: `${item.key}-duplicate` };
+        }
+        return item;
+    });
+};
+
+const informationCarouselContent: InformationCarouselContent[] = [
+    {
+        title: 'Algorithm',
+        description: 'Users going through a vetting process to ensure you never match with bots.',
+    },
+    {
+        title: 'Matches',
+        description: 'We match you with people that have a large array of similar interests.',
+    },
+    {
+        title: 'Premium',
+        description: 'Sign up today and enjoy the first month of premium benefits on us.',
+    },
+];
+
+function Onboarding() {
+    const screenWidth = screen.width;
+    const ITEM_SIZE = screenWidth * 0.6;
+    const SPACER_SIZE = (screenWidth - ITEM_SIZE) / 2;
+    const scrollX = useRef(new Animated.Value(0)).current;
+
+    const infoCarouselRef = useRef<FlatList<InformationCarouselContent> | null>(null);
+    const [activeInfoIndex, setActiveInfoIndex] = useState(0);
+    const flatListRef = useRef<FlatList>(null);
+    const [imageData, setImageData] = useState(duplicateDataWithUniqueKeys());
```

Comment:
We could just use useRef to store image data as we won't update it

Reply:
alright sounds fine to me